pkg/operator/ceph/cluster/crash/crash_test.go

```diff
@@ -17,13 +17,90 @@ limitations under the License.
 package crash
 
 import (
+	"context"
 	"testing"
 
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned/fake"
+	"github.com/rook/rook/pkg/client/clientset/versioned/scheme"
+	"github.com/rook/rook/pkg/clusterd"
+	cephver "github.com/rook/rook/pkg/operator/ceph/version"
+	"github.com/rook/rook/pkg/operator/test"
 	"github.com/stretchr/testify/assert"
+	"github.com/tevino/abool"
+	v1 "k8s.io/api/batch/v1"
+	"k8s.io/api/batch/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	cntrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 func TestGenerateCrashEnvVar(t *testing.T) {
 	env := generateCrashEnvVar()
 	assert.Equal(t, "CEPH_ARGS", env.Name)
 	assert.Equal(t, "-m $(ROOK_CEPH_MON_HOST) -k /etc/ceph/crash-collector-keyring-store/keyring", env.Value)
 }
+
+func TestCreateOrUpdateCephCron(t *testing.T) {
+	cephCluster := cephv1.CephCluster{ObjectMeta: metav1.ObjectMeta{Namespace: "rook-ceph"}}
+	cephVersion := &cephver.CephVersion{Major: 16, Minor: 2, Extra: 0}
+	ctx := context.TODO()
+	context := &clusterd.Context{
+		Clientset:                  test.New(t, 1),
+		RookClientset:              rookclient.NewSimpleClientset(),
+		RequestCancelOrchestration: abool.New(),
+	}
+
+	s := scheme.Scheme
+	err := v1.AddToScheme(s)
+	if err != nil {
+		assert.Fail(t, "failed to build scheme")
+	}
+	err = v1beta1.AddToScheme(s)
+	if err != nil {
+		assert.Fail(t, "failed to build scheme")
+	}
+
+	r := &ReconcileNode{
+		scheme:  s,
+		client:  fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects().Build(),
+		context: context,
+	}
+
+	cronV1 := &v1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prunerName,
+			Namespace: "rook-ceph",
+		},
+	}
+
+	cronV1Beta1 := &v1beta1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prunerName,
+			Namespace: "rook-ceph",
+		},
+	}
+
+	// check if v1beta1 cronJob is present and v1 cronJob is not
+	controllerutil, err := r.createOrUpdateCephCron(cephCluster, cephVersion, false)
+	assert.NoError(t, err)
+	assert.Equal(t, controllerutil, cntrlutil.OperationResult("created"))
+
+	err = r.client.Get(ctx, types.NamespacedName{Namespace: "rook-ceph", Name: prunerName}, cronV1Beta1)
+	assert.NoError(t, err)
+
+	err = r.client.Get(ctx, types.NamespacedName{Namespace: "rook-ceph", Name: prunerName}, cronV1)
+	assert.EqualError(t, err, "cronjobs.batch \"rook-ceph-crashcollector-pruner\" not found")
```

Comment:
It would be good option to check for IsNotFound here rather than checking the error string. 

```
assert.Error(t, err)
assert.True(t, kerrors.IsNotFound(err))
```

same in line 105 below.

Reply:
done.